backend/internal/authn/assert/assurance.go

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright (c) 2025, WSO2 LLC. (https://www.wso2.com).
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package assert
+
+import (
+	authncm "github.com/asgardeo/thunder/internal/authn/common"
+)
+
+// assuranceLevelOrder maps assurance levels to their hierarchical order for comparison.
+var assuranceLevelOrder = map[AssuranceLevel]int{
```

Comment:
Rather keeping another map, can't we improve the `AssuranceLevel` object itself to keep this level value?

Reply:
This got removed with the latest changes